### DISCUSSION TITLE
WSL 1/2 可以进行快照吗？

### DISCUSSION THREAD

**Discussion by heroxbd** [physics-data/faq#53] [Q&A] [answered]
在升级 WSL 内的发行版之前，我想做一个快照，万一跪了能够恢复。

看到 Reddit 上的讨论，似乎 WSL 根本没这功能：

  https://www.reddit.com/r/bashonubuntuonwindows/comments/p5qu3a/running_wsl_2_on_windows_in_a_vm/

真的是这样吗？只能把整个 Windows 10 来个快照才行？

**Comment by jiegec** [comment, marked as answer]
可以导出 rootfs：https://winaero.com/export-import-wsl-linux-distro-windows-10/

**Reply by heroxbd** [reply]
有道理！

**Comment by Berrysoft** [comment]
WSL2的备份更简单，它本身是个虚拟机，所有文件在一个vhdx里面，复制一份就行。
